### PR TITLE
WeakMap and WeakSet

### PR DESCRIPTION
**Pull request by mahdiHashemi14**
Please review and comment / commit if anything is wrong

Summary of article needs a little more attention

I didn't know how to translate "flags" in the title of "recipients read" task, best to take a look

### CODE REVIEW

File: 1-js/05-data-types/08-weakmap-weakset/article.md [hunk 1]
```diff
@@ -259,33 +260,33 @@ let john = { name: "John" };
...
+// به طور خودکار از آن شیء تمیز می‌شود visitedSet
 ```
 
-The most notable limitation of `WeakMap` and `WeakSet` is the absence of iterations, and the inability to get all current content. That may appear inconvenient, but does not prevent `WeakMap/WeakSet` from doing their main job -- be an "additional" storage of data for objects which are stored/managed at another place.
+بزرگ‌ترین محدودیت `WeakMap` و `WeakSet` این است که نمی‌توان در آن حلقه زد و تمام محتوای حال حاضر را گرفت. این موضوع ممکن است اذیت کننده باشد اما باعث نمی‌شود که `WeakMap/WeakSet` کار اصلی خودشان را انجام ندهند -- اینکه یک حافظه «اضافیِ» داده، برای شیءهایی باشند که در جایی دیگر ذخیره/مدیریت می‌شوند.
 
 ## Summary
 
-`WeakMap` is `Map`-like collection that allows only objects as keys and removes them together with associated value once they become inaccessible by other means.
+`WeakMap` یک مجموعه شبیه `Map` است که فقط شیءها را به عنوان کلید قبول می‌کند و همان شیء و داده‌های تخصیص داده شده به آن را زمانی که شیء از طرق دیگر غیر قابل دسترس شود، حذف می‌کند.
```
Comment (mahdyar) [inline 1-js/05-data-types/08-weakmap-weakset/article.md:284]:
```suggestion
`WeakMap` یک مجموعه `Map`-like است که فقط شیءها را به عنوان کلید قبول می‌کند و همان شیء و داده‌های تخصیص داده شده به آن را زمانی که شیء از طرق دیگر غیر قابل دسترس شود، حذف می‌کند.
```

File: 1-js/05-data-types/08-weakmap-weakset/article.md [hunk 2]
```diff
@@ -259,33 +260,33 @@ let john = { name: "John" };
...
 
-`WeakSet` is `Set`-like collection that stores only objects and removes them once they become inaccessible by other means.
+`WeakSet` یک مجموعه شبیه `Set` است که فقط شیءها را ذخیره می‌کند و زمانی که آنها از طرق دیگر غیر قابل دسترس شوند، حذف‌شان می‌کند.
```
Comment (mahdyar) [inline 1-js/05-data-types/08-weakmap-weakset/article.md:286]:
```suggestion
`WeakSet` یک مجموعه `Set`-like است که فقط شیءها را ذخیره می‌کند و زمانی که آنها از طرق دیگر غیر قابل دسترس شوند، حذف‌شان می‌کند.
```

File: 1-js/05-data-types/08-weakmap-weakset/01-recipients-read/task.md [hunk 3]
```diff
@@ -2,9 +2,9 @@ importance: 5
 
 ---
 
-# Store "unread" flags
+# پیام‌های «خوانده نشده» را ذخیره کنید
```
Comment (mahdyar) [inline 1-js/05-data-types/08-weakmap-weakset/01-recipients-read/task.md:5]:
```suggestion
# پرچم‌های «خوانده نشده» را ذخیره کنید
```

**Review by mahdyar — CHANGES_REQUESTED** [review]

**Comment by mahdiHashemi14** [comment]
Thank you, @mahdyar 
/done

**Review by mahdyar — APPROVED** [review]
Thank you, @mahdiHashemi14!

**Pull request merged**